src/general/ModLoader.cs

```diff
@@ -0,0 +1,320 @@
+            file.Close();
+
+            return result;
+        }
+    }
+
+    private void OnUnloadedModSelected(int index)
+    {
+        var tempModInfo = modList[index];
+        modInfoName.Text = tempModInfo.ModName;
+        modInfoAuthor.Text = "Author: " + tempModInfo.Author;
+        modInfoVersion.Text = "Version: " + tempModInfo.Version;
+        modInfoDescription.Text = tempModInfo.Description;
+        modInfoDescription.Text = tempModInfo.Description;
+        loadedItemList.UnselectAll();
+    }
+
+    private void OnLoadedModSelected(int index)
+    {
+        var tempModInfo = loadedModList[index];
+        modInfoName.Text = tempModInfo.ModName;
+        modInfoAuthor.Text = "Author: " + tempModInfo.Author;
+        modInfoVersion.Text = "Version: " + tempModInfo.Version;
+        modInfoDescription.Text = tempModInfo.Description;
+        modInfoDescription.Text = tempModInfo.Description;
+        unloadedItemList.UnselectAll();
+    }
+
+    private void OnMoveToLoadPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        if (unloadedItemList.GetSelectedItems().Length <= 0)
+        {
+            return;
+        }
+
+        var selectedItem = unloadedItemList.GetSelectedItems()[0];
+        loadedModList.Add(modList[selectedItem]);
+        loadedItemList.AddItem(modList[selectedItem].ModName);
+        modList.RemoveAt(selectedItem);
+        unloadedItemList.RemoveItem(selectedItem);
+    }
+
+    private void OnMoveToUnloadPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        if (loadedItemList.GetSelectedItems().Length <= 0)
+        {
+            return;
+        }
+
+        var selectedItem = loadedItemList.GetSelectedItems()[0];
+        modList.Add(loadedModList[selectedItem]);
+        unloadedItemList.AddItem(loadedModList[selectedItem].ModName);
+        loadedModList.RemoveAt(selectedItem);
+        loadedItemList.RemoveItem(selectedItem);
+    }
+
+    private void OnRefreshPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        modList.Clear();
+        loadedModList.Clear();
+        unloadedItemList.Clear();
+        loadedItemList.Clear();
+        DirectoryInfo modFolder = Directory.CreateDirectory($"{Directory.GetCurrentDirectory()}\\mods");
+        foreach (DirectoryInfo currentMod in modFolder.EnumerateDirectories())
+        {
+            if (!File.Exists(currentMod.FullName + "/mod_info.json"))
+            {
+                continue;
+            }
+
+            var currentModInfo =
+                JsonConvert.DeserializeObject<ModInfo>(ReadJSONFile(currentMod.FullName + "/mod_info.json"));
+
+            if (currentModInfo.AutoLoad)
+            {
+                continue;
+            }
+
+            currentModInfo.Location = currentMod.FullName;
+            modList.Add(currentModInfo);
+            unloadedItemList.AddItem(currentModInfo.ModName);
+        }
+    }
+
+    private void OnLoadPressed()
```

Comment:
What happens if user presses this multiple times? Does this work? Do DLL containing mods cause issues? What if we ever add callbacks for mods to react to loading?
I think this needs some changes to not load already loaded mods.